cli/translator.js

```diff
@@ -0,0 +1,58 @@
+import fs from 'fs/promises'
+
+export default async function translator(path) {
+  const cwd = process.cwd()
+  const rootPath = await searchRootFolder(cwd)
+  if (!rootPath) {
+    throw Error(`Cannot translate using ${cwd} folder`)
+  }
+
+  const relativeRoot = removeAppFromPath(cwd.substr(rootPath.length))
+  const isAppPath = path?.startsWith(`app`)
+  const resolvePath = isAppPath 
+    ? removeAppFromPath(path) 
+    : path ?? ``
+
+  const translatedPath = isAppPath
+    ? resolvePath
+    : `${relativeRoot ? relativeRoot + `/` : ``}` + resolvePath
+
+  return translatedPath
+}
+
+// translate app/a/b/c => a/b/c
+function removeAppFromPath(path) {
+  return path
+    .split(`/`)
+    .filter(dir => dir.length > 0 && dir !== `app`)
+    .join(`/`)
+}
```

Comment:
I think there might be a small bug in this code. Consider this scenario:

- A nano has been named `app` (it's a poor choice, but legal)
- Path `root/app/east/asia/nanos/app/demo.js`
- CWD `root/app/east/asia/nanos/app/`

```
removeAppFromPath(`app/east/asia/nanos/app/`)

// Result: `east/asia/nanos/`
// Expected: `east/asia/nanos/app/
```

Reply:
DONE